clusterDeployer.py

```diff
@@ -221,6 +221,25 @@ def teardown(self) -> None:
 
         print(lh.run(f"ip link set eno1 nomaster"))
 
+    def _validate_external_port(self, lh):
+        # external interface only needed with BM or BF hosts
+        # validation is succesful, but the name is irrelevant
+        if all(x["type"] == "vm" for x in self._cc["workers"]):
+            return True
+
+        # do automatic detection, if needed
+        if self._cc["exteranl_port"] == "auto":
+            self._cc["external_port"] = common.extract_port(lh.run("ip -json r").out, "default")
```

Comment:
Let's start moving these ip calls into host.py.

Reply:
Also, the auto-detection now seems to use the port that has the default uplink. Since the API network should _not_ be on the same network, the default should actually be any port that's _not_ the default route and that has connectivity.